calendar/tests/test.bal

```diff
@@ -0,0 +1,421 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/log;
+import ballerina/test;
+import ballerina/os;
+
+configurable string refreshUrl = os:getEnv("REFRESH_URL");
+configurable string refreshToken = os:getEnv("REFRESH_TOKEN");
+configurable string clientId = os:getEnv("CLIENT_ID");
+configurable string clientSecret = os:getEnv("CLIENT_SECRET");
+
+Configuration configuration = {
+    clientConfig: {
+        refreshUrl: refreshUrl,
+        refreshToken : refreshToken,
+        clientId : clientId,
+        clientSecret : clientSecret
+    }
+};
+
+Client calendarClient = check new(configuration);
+
+string eventId  = "";
+string defaultCalendarId = "";
+string specificCalendarId = "";
+string queryParamSelect = "$select=subject"; 
+string queryParamTop = "$top=5";
+string queryParamCount = "$count=true";
+
+// Tests related to `Event` resource operations
+@test:Config {
+    enable: true,
+    groups: ["events"],
+    before: testCreateEvent,
+    after: testDeleteEvent
+}
+function testGetEvent() {
+    log:printInfo("client->testGetEvent()"); 
+    Event|error event = calendarClient->getEvent(eventId);
+    if(event is Event) {
+        test:assertEquals(event.id, eventId, "Invalid Event ID");
+        log:printInfo("Event received with ID : " + event.id.toString());
+    } else {
+        test:assertFail(msg = event.message());
+    }
+    io:println("\n\n");
+}
+
+@test:Config {
+    enable: true,
+    groups: ["events"],
+    before: testCreateEvent,
+    after: testDeleteEvent
+}
+function testGetEventWithPreferenceHeaders() {
+    log:printInfo("client->testGetEventWithPreferenceHeaders()"); 
+    Event|error event = calendarClient->getEvent(eventId, timeZone=TIMEZONE_AD, contentType=CONTENT_TYPE_TEXT);
+    if(event is Event) {
+        test:assertEquals(event.id, eventId, "Invalid Event ID");
+        log:printInfo("Event received with requested timezone : " + event?.'start?.timeZone.toString());
+    } else {
+        test:assertFail(msg = event.message());
+    }
+    io:println("\n\n");
+}
+
+@test:Config {
+    enable: true,
+    groups: ["events"],
+    before: testCreateEvent,
+    after: testDeleteEvent
+}
+function testGetEventWithQueryParameters() {
```

Comment:
It will be worth to describe what we test here in one sentence - wDYT?

Reply:
Yeah. It is worth discribing. Lets do

Reply:
Added in all test cases
/Thanks